### ISSUE TITLE
[Transform] SourceDestValidatorTests.testRemoteSourceDoesNotExist test failure

### ISSUE THREAD

**Issue by hendrikmuhs** [elastic/elasticsearch#50420]
Fails on 7.x:

```
./gradlew ':x-pack:plugin:core:test' --tests "org.elasticsearch.xpack.core.common.validation.SourceDestValidatorTests.testRemoteSourceDoesNotExist" -Dtests.seed=5105F6CB1E897072 -Dtests.security.manager=true -Dtests.locale=en-GB -Dtests.timezone=Asia/Colombo -Dcompiler.java=13 -Druntime.java=8
```

Test is muted

**Comment by hendrikmuhs** [comment]
The problem is that the following PR is only on master #40419 - the corresponding issue #37863 indicates that this is by design, because it's a breaking change.

Looks like we need a work-around for 7.x

**Comment by hendrikmuhs** [comment]
fixed by #50460
